External/Plugins/ASCompletion/Completion/ASGenerator.cs

```diff
@@ -2841,72 +2841,91 @@ private static void GenerateFunctionJob(GeneratorJobType job, ScintillaControl s
             if (caller?.Parameters != null && caller.Parameters.Count > 0)
             {
                 Func<string, string> cleanType = null;
-                cleanType = s => s.StartsWith("(") && s.EndsWith(')') ? cleanType(s.Trim('(', ')')) : s;
+                cleanType = s => s.StartsWith("(") && s.EndsWith(')') ? cleanType(s.Trim('(', ')')) : s;
```

Comment:
I don't understand why the recursion is needed, since `Trim('(', ')')` would already make it that it doesn't start or end with '(' and ')'.

Maybe it's meant to be
```c#
s => s.StartsWith('(') && s.EndsWith(')') ? cleanType(s.Substring(1, s.Length - 2)) : s
```
or better without the recursion or useless string creations
```
s => {
    int start = 0, end = s.Length - 1;
    while (start < end && s[start] == '(' && s[end] == ')') {
        start++; end--;
    }
    return s.Substring(start, end + 1 - start);
}
```

Why is it a local delegate instead of a method though?